pkg/state/license.go

```
@@ -0,0 +1,144 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+
+	memdb "github.com/hashicorp/go-memdb"
+	"github.com/kong/go-database-reconciler/pkg/utils"
+)
+
+const (
+	licenseTableName = "license"
+)
+
+var licenseTableSchema = &memdb.TableSchema{
+	Name: licenseTableName,
+	Indexes: map[string]*memdb.IndexSchema{
+		"id": {
+			Name:    "id",
+			Unique:  true,
+			Indexer: &memdb.StringFieldIndex{Field: "ID"},
+		},
+		all: allIndex,
+	},
+}
+
+type LicensesCollection collection
+
+func getLicense(txn *memdb.Txn, id string) (*License, error) {
+	res, err := multiIndexLookupUsingTxn(txn, licenseTableName, []string{"id"}, id)
+	if err != nil {
+		return nil, err
+	}
+	l, ok := res.(*License)
+	if !ok {
+		panic(unexpectedType)
```

Why not return an error?

---

This is kept the same as other types.